### PR TITLE
fix(settings): Fix organization invite members - team selector

### PR DESCRIPTION
**Pull request by billyvg**
This changes the team selectors to not use teams from TeamsStore via `withTeams`, instead load teams via API and require users to filter down teams. Limit teams to 25 for speed.

See https://github.com/getsentry/sentry/pull/18177 for more information.

Fixes JAVASCRIPT-2206

### CODE REVIEW

**Review by evanpurkhiser — APPROVED** [review]
Nice, looks pretty good to me.

File: tests/js/spec/views/settings/organizationMembers/inviteRequestRow.spec.jsx [hunk 1]
```diff
@@ -133,23 +139,25 @@ describe('InviteRequestRow', function() {
...
+
     selectByValue(wrapper, 'member', {name: 'role', control: true});
     expect(mockUpdate).toHaveBeenCalledWith({role: 'member'});
 
     selectByValue(wrapper, 'one', {name: 'teams', control: true});
     expect(mockUpdate).toHaveBeenCalledWith({teams: ['one']});
   });
 
-  it('cannot be approved when invitee role is not allowed', function() {
+  it('cannot be approved when invitee role is not allowed', async function() {
```
Comment (priscilawebdev) [inline tests/js/spec/views/settings/organizationMembers/inviteRequestRow.spec.jsx:160]:
```suggestion
  it('cannot be approved when invite role is not allowed', async function() {
```

File: src/sentry/static/sentry/app/components/modals/inviteMembersModal/index.tsx [hunk 2]
```diff
@@ -359,12 +357,7 @@ class InviteMembersModal extends AsyncComponent<Props, State> {
               )
             }
             onChangeRole={({value}) => this.setRole(value, i)}
-            onChangeTeams={opts =>
-              this.setTeams(
-                opts.map(v => v.value),
-                i
-              )
-            }
+            onChangeTeams={opts => this.setTeams(opts ? opts.map(v => v.value) : [], i)}
```
Comment (priscilawebdev) [inline src/sentry/static/sentry/app/components/modals/inviteMembersModal/index.tsx:360]:
```suggestion
            onChangeTeams={opts => this.setTeams(opts?.map(v => v.value) ?? [], i)}
```

File: src/sentry/static/sentry/app/utils/loadTeamsForSelectOptions.tsx [hunk 3]
```diff
@@ -0,0 +1,18 @@
+import {Client} from 'app/api';
+
+/**
+ * Used for async `react-select` component to choose and filter teams
+ */
+function loadTeamsForSelectOptions(orgSlug: string) {
+  const api = new Client();
+
+  return async (inputValue: string) => {
+    const resp = await api.requestPromise(`/organizations/${orgSlug}/teams/`, {
```
Comment (priscilawebdev) [inline src/sentry/static/sentry/app/utils/loadTeamsForSelectOptions.tsx:10]: shouldn't we use a` try ... catch` here?

**Pull request closed without merging**